ast.ts

```diff
@@ -25,6 +25,7 @@ export type Stmt<A> =
   | {  a?: A, tag: "while", cond: Expr<A>, body: Array<Stmt<A>> }
   | {  a?: A, tag: "pass" }
   | {  a?: A, tag: "field-assign", obj: Expr<A>, field: string, value: Expr<A> }
+  | {  a?: A, tag: "lookup-assign", name: string, key: Expr<A>, value: Expr<A> }
```

Comment:
As discussed in #9 here's a quick link to an idea of the interface we've proposed for fully unifying assignment. Note: we're still using the old `id-assign` and `field-assign` parsing for this initial PR. https://github.com/ucsd-cse231-w21/chocopy-wasm-compiler/blob/11cb027e707adaa4c9667bbaf0bf5d8f740e9040/proposals/destructuring.md#why-this-form-of-unified-assignable

Reply:
Thanks! I removed the assign statement from the ast for now. We will wait for your changes to get in, and then use the interface that you've proposed.

Reply:
Awesome! I'll make sure to tag in once we have something in a subsequent PR.